File: src/main/java/com/gl/module/system/domain/SysMenu.java
package com.gl.module.system.domain;

import com.gl.infra.spring.data.domain.IdEntity;

import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * @author gantrylau
 * @since 2017/2/21
 */
@Entity
@Table(name = "sys_menu")
public class SysMenu extends IdEntity<Long> {

    private String name;

    private String alias;

    private Integer sort;

    private String parentAlias;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAlias() {
        return alias;
    }

    public void setAlias(String alias) {
        this.alias = alias;
    }

    public Integer getSort() {
        return sort;
    }

    public void setSort(Integer sort) {
        this.sort = sort;
    }

    public String getParentAlias() {
        return parentAlias;
    }

    public void setParentAlias(String parentAlias) {
        this.parentAlias = parentAlias;
    }
}

File: src/main/java/com/gl/module/system/domain/SysMenu.kt
package com.gl.module.system.domain

import com.gl.infra.spring.data.domain.IdEntity
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

/**
 * @author gantrylau
 * *
 * @since 2017/1/27
 */
@Entity
@Table(name = "sys_menu")
open class SysMenu : IdEntity<Long>() {

    open var name: String? = null

    open var alias: String? = null

    open var sort: Int? = null

    open var parentAlias: String? = null

    open var isLeaf: Boolean = false
        @Column(name = "is_leaf", nullable = false)
        get
}
